### PR TITLE
Support for containerd, use cni everywhere

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

This is a shift to CNI by default.
This allows also to start with a different container runtimes, in this PR it's containerd and docker.

### Motivation

Evolve to a more generic Kubernetes setup.

### Additional Notes

This is still work in progress.

### CODE REVIEW

File: pkg/setup/network.go [hunk 1]
```diff
@@ -274,11 +258,16 @@ func (e *Environment) setupNetwork() error {
...
-		e.isDockerBridge = true
-		return nil
+	err = e.generateCNIConf()
+	if err != nil {
+		return err
 	}
-	e.isDockerBridge = false
 
+	// docker set a drop by default
+	b, err := exec.Command("iptables", "-P", "FORWARD", "ACCEPT").CombinedOutput()
```
Comment (lbernail) [inline pkg/setup/network.go]: If you know the name of the CNI bridge you can add explicitly add it to the FORWARD chain with ACCEPT (look at the docker0 config)

**Comment by JulienBalestra** [comment]
This is working fine and ready for final review @DataDog/container-integrations

File: cmd/cli/cmd.go [hunk 2]
```diff
@@ -299,6 +299,15 @@ func NewCommand() (*cobra.Command, *int) {
 	daemonCommand.PersistentFlags().String("kubeconfig-path", config.ViperConfig.GetString("kubeconfig-path"), "path to the kubeconfig file")
 	config.ViperConfig.BindPFlag("kubeconfig-path", daemonCommand.PersistentFlags().Lookup("kubeconfig-path"))
 
+	daemonCommand.PersistentFlags().String("kubernetes-cluster-ip-range", config.ViperConfig.GetString("kubernetes-cluster-ip-range"), "kubernetes cluster CIDR")
+	config.ViperConfig.BindPFlag("kubernetes-cluster-ip-range", daemonCommand.PersistentFlags().Lookup("kubernetes-cluster-ip-range"))
+
+	daemonCommand.PersistentFlags().String("pod-ip-range", config.ViperConfig.GetString("pod-ip-range"), "pod common network interface CIDR")
+	config.ViperConfig.BindPFlag("pod-ip-range", daemonCommand.PersistentFlags().Lookup("pod-ip-range"))
+
+	daemonCommand.PersistentFlags().String("container-runtime", config.ViperConfig.GetString("container-runtime"), `container runtime interface to use (experimental: "containerd"), leave empty to use docker`)
```
Comment (devonboyer) [inline cmd/cli/cmd.go]: This could just say "the default is docker"

File: pkg/config/config.go [hunk 3]
```diff
@@ -25,17 +25,25 @@ const (
...
 
 func init() {
 	ViperConfig.SetDefault("hyperkube-version", "1.10.3")
 	ViperConfig.SetDefault("vault-version", "0.9.5")
 	ViperConfig.SetDefault("etcd-version", "3.1.11")
 	ViperConfig.SetDefault("cni-version", "0.7.0")
+	ViperConfig.SetDefault("containerd-version", "1.1.0")
+	ViperConfig.SetDefault("runc-version", "1.0.0-rc5")
+
+	ViperConfig.SetDefault("container-runtime", "")
```
Comment (devonboyer) [inline pkg/config/config.go]: Should the default `container-runtime` not be docker?

File: pkg/setup/network.go [hunk 4]
```diff
@@ -267,18 +251,32 @@ func (e *Environment) setupNetwork() error {
...
-	if isDockerBridge() {
-		e.isDockerBridge = true
-		return nil
+	err = e.generateCNIConf(defaultBridgeName)
+	if err != nil {
+		return err
 	}
-	e.isDockerBridge = false
 
+	// docker set a drop by default
```
Comment (devonboyer) [inline pkg/setup/network.go]: I'm not really sure what this comment means.

**Pull request merged**